crates/node-bindings/src/parcel/worker.rs

```diff
@@ -1,10 +1,20 @@
+use std::sync::mpsc::Sender;
+
 use napi::{Env, JsObject, JsUndefined};
 use napi_derive::napi;
 use parcel::rpc::nodejs::NodejsWorker;
+use parcel_napi_helpers::JsTransferable;
 
 #[napi]
-pub fn register_worker(env: Env, worker: JsObject) -> napi::Result<JsUndefined> {
+pub fn register_worker(
+  env: Env,
+  channel: JsTransferable<Sender<NodejsWorker>>,
+  worker: JsObject,
+) -> napi::Result<JsUndefined> {
   let worker = NodejsWorker::new(worker)?;
-  NodejsWorker::register_worker(worker);
+  let tx_worker = channel.take()?;
+  if tx_worker.send(worker).is_err() {
```

Comment:
I don't think it's safe to send this across threads - https://github.com/parcel-bundler/parcel/blob/445b9bc0ca57922a20977ee81b50e7a07e35b6ab/crates/parcel_plugin_rpc/src/nodejs/rpc_conn_nodejs.rs#L11

Reply:
This is safe due to `JsCallable` being thread-safe